pkg/controllers/core/catalog_controller.go

```diff
@@ -217,6 +249,28 @@ func updateStatusUnpackFailing(status *v1alpha1.CatalogStatus, err error) error
 	return err
 }
 
+func updateStatusStorageError(status *v1alpha1.CatalogStatus, err error) error {
+	status.ResolvedSource = nil
+	status.Phase = v1alpha1.PhaseFailing
+	meta.SetStatusCondition(&status.Conditions, metav1.Condition{
+		Type:    v1alpha1.TypeStorage,
```

Comment:
No need for a new condition type.
```suggestion
		Type:    v1alpha1.TypeUnpacked,
```